umi_tools/umi_methods.py

```diff
@@ -1104,6 +1104,80 @@ def get_read_position(read, soft_clip_threshold):
     return start, pos, is_spliced
 
 
+class get_readpairs:
+    # Used in place of a real read as the mate of reads that aren't properly paired
+    NoMate = object()
+
+    def incomplete_pair_to_singleton(self, queue_entry):
+        for i in [i for i, v in enumerate(queue_entry) if v is get_readpairs.NoMate]:
+            queue_entry[i] = None
+
+    def incomplete_pairs_as_singletons(self):
+        self.incomplete_queue_entries.clear()
+        while self.queue:
+            queue_entry = self.queue.popleft()
+            if get_readpairs.NoMate in queue_entry:
+                # Incomplete entry. Turn into singleton entry
+                read = next(r for r in queue_entry if r is not get_readpairs.NoMate)
+                U.warn("inconsistent BAM file: only one of the two mates of proper pair %s found on chromosome %s, treating as unpaired" %
```

Comment:
I think that this is probably going to be very common for people that are mapping to the transcriptome, where each transcript is a different contig.

Reply:
Ok, but in this case does --paired mode make any sense at all?